drivers/net/dsa/lantiq_gsw_mdio.c

```diff
@@ -35,19 +36,103 @@
 #include "lantiq_gsw.h"
 #include "lantiq_pce.h"
 
-static u32 gsw_mdio_read(struct gswip_priv *priv, void *address)
+#define NUM_ACCESSIBLE_REGS (30)
+#define TARGET_BASE_ADDRESS_REG (31)
+
+struct gsw_mdio {
+	struct mdio_device *mdio_dev;
+	struct gswip_priv common;
```

Comment:
This is currently how I've restructured the driver data:
The "driver data" pointer that gets registered with the kernel (i.e. the private pointer in the device structure) now points at these bus-specific `gsw_mdio` and `gsw_platform` structures containing the common `gswip_priv` data as well as a pointer to the specific specific-device objects necessary for some of their operations. It is likely they will get filled out with more data specific to each bus as I move into implementing the probe functions.

An alternative would have been to keep the "driver data" pointer as `gswip_priv` and add a `void*` to that structure to allow access to bus-specific data. That may end up being cleaner, I will re-evaluate after the probe functions are written. For now I avoided that solution as using a `void*` to hold pointers to different structures depending on the specific instantiation of the driver is not super intuitive and I'm not sure what the kernel community's stance is on it.